iris-mpc/src/bin/server.rs

```diff
@@ -1012,6 +1016,66 @@ async fn server_main(config: Config) -> eyre::Result<()> {
     let db_chunks_bucket_name = config.db_chunks_bucket_name.clone();
     let db_chunks_folder_name = config.db_chunks_folder_name.clone();
 
+    // --------------------------------------------------------------------------
+    // ANCHOR: Syncing latest node state
+    // --------------------------------------------------------------------------
+    tracing::info!("⚓️ ANCHOR: Syncing latest node state");
+    let all_nodes = config.node_hostnames.clone();
+    let next_node = &all_nodes[(config.party_id + 1) % 3];
+    let prev_node = &all_nodes[(config.party_id + 2) % 3];
+
+    tracing::info!("Database store length is: {}", store_len);
+    let mut states = vec![my_state.clone()];
+    for host in [next_node, prev_node].iter() {
+        let res = reqwest::get(format!("http://{}:3000/sync", host)).await;
+        match res {
+            Ok(res) => {
+                let state: SyncState = match res.json().await {
+                    Ok(state) => state,
+                    Err(e) => {
+                        tracing::error!("Failed to parse sync state from party {}: {:?}", host, e);
+                        panic!(
+                            "could not get sync state from party {}, trying to restart",
+                            host
+                        );
+                    }
+                };
+                states.push(state);
+            }
+            Err(e) => {
+                tracing::error!("Failed to fetch sync state from party {}: {:?}", host, e);
+                panic!(
+                    "could not get sync state from party {}, trying to restart",
+                    host
+                );
+            }
+        }
+    }
+    let sync_result = SyncResult::new(my_state, states);
+
+    if let Some(db_len) = sync_result.must_rollback_storage() {
+        tracing::error!("Databases are out-of-sync: {:?}", sync_result);
+        if db_len + max_rollback < store_len {
+            return Err(eyre!(
+                "Refusing to rollback so much (from {} to {})",
+                store_len,
+                db_len,
+            ));
+        }
+        tracing::warn!(
+            "Rolling back from database length {} to other nodes length {}",
+            store_len,
+            db_len
+        );
+        store.rollback(db_len).await?;
+        metrics::counter!("db.sync.rollback").increment(1);
+    }
+
+    if download_shutdown_handler.is_shutting_down() {
+        tracing::warn!("Shutting down has been triggered");
+        return Ok(());
+    }
```

Comment:
is this connected to the rollback or the db loading?
i.e., should this be here or left with the DB loading procedure

Reply:
should be fine to leave it here with pg consistency?